### DISCUSSION TITLE
KitQL & Houdini

### DISCUSSION THREAD

**Discussion by jycouet** [jycouet/kitql#156] [Announcements]
For the past 2 months, @AlecAivazis and I have started to figure out what would be the best GraphQL EcoSystem in SvelteKit? We both have a great library working on similar topics but not tackling the thing the same way. 

We decided to join forces and provide THE best experience for the community, for you 🥳

## In short
- `KitQL` contains all that you can dream of when you speak about SvelteKit & GraphQL... client (Houdini), server (Yoga), modules, linting, best practices, ... and more!
- `Houdini` is THE client-side of SvelteKit & GraphQL, managing operations, stores, inline, caching, optimistic UI, pagination, ... and more!
- How it look like? 🎨🪄👇![2022 05 25 - KitQL   Houdini](https://user-images.githubusercontent.com/5312607/173023713-349366b0-e2c2-4742-b4c8-130fe91f1a4a.gif)

## Next? 
To continue the story, we are proceeding in steps (in public!)
🔵 Store API in Houdini https://github.com/HoudiniGraphql/houdini/pull/291
🔵 Houdini Preview
🔵 Tweak docs
🔵 Adjusting KitQL workflow
🔵 Official release of KitQL & Houdini working together

We are confident now that this will happen in the coming weeks and will start communicating more about it.

🙏 Your feedback is very much appreciated.

**Comment by frederikhors** [comment]
I am the architect of this union!

Congratulations guys! Can't wait to try and give you feedback!

**Reply by gevera** [reply]
Wow. I was about to switch to Houdini for the lack of Subscriptions in KitQL. Now this union of Houdini and KitQL is perfect. Looking forward for the release!

**Comment by giray123** [comment]
Hi, KitQL looks awesome 🙏  and we have decided to use it for our company product although it is in the early stages. We thought may be it is a good idea to wait for the Houdini integration, because it might change the structure quite a bit. Do you have an estimation about when the integration will be released so that we can also plan for our migration?

**Reply by jycouet** [reply]
Thank you for your nice comment. We are working hard for you guys 🪄 🥳
It's complicated for us to give you a date as we want to cover a lot of use cases and have an easy migration from both sides.
For sure it's "as soon as possible".

We will keep you posted

**Comment by choudel** [comment]
I like kitql , I knew about it  because I had problems with Urql , with kitql I didn't had similar problems it was smooth sailing , the thing is my graphql schema is generated with postgraphile and it's massive what could you suggest for me ?  also i'm not very pro lol

**Reply by jycouet** [reply]
No worries. You will be able to do the same thing 👌
And we will still point to your schema (generated or not)

**Comment by mingsterism** [comment]
From looking at the Houdini docs, i looks like KitQL is totally integrated into Houdini and there's no mention of KitQL. 
Ultimately would KitQL be replaced by houdini, and should we just use houdini? 
![image](https://user-images.githubusercontent.com/12984614/175841967-0cb5502f-9271-4550-9856-0dc5ae59f54f.png)

**Comment by AlecAivazis** [comment]
Good question @mingsterism! If you pardon me speaking for @jycouet i might be able to answer your questions. 

So right away let me say that KitQL is still going very strong. I know @jycouet has _many_ plans for KitQL and I’m really looking forward to helping make them real. This project has always been about providing the best possible “batteries included” solution for full-stack GraphQL in SvelteKit and it will absolutely continue to do that. The only real change moving forward is that the client side of a KitQL project will use Houdini for fetching and mutating data. 

KitQL brings a lot to the table to streamline your dev process so I wouldn’t think of Houdini as a replacement for KitQL. Instead it’s more like Houdini is becoming part of the KitQL Stack.

**Reply by jycouet** [reply]
Perfect answer @AlecAivazis 🎉 

And, to hype a bit more people, let's reveal our **planned** release date for everything mentioned in the original discussion
    👉🗓️ `4th of July 2022` !!!
    
Until then we
- migrate a few real projects _for final testing_.
- prepare the communication
- wait for all your support 🥳

**Reply by letmejustputthishere** [reply]
very much looking forward to that!

**Reply by nikfp** [reply]
Super excited to see what you all came up with!

**Reply by mingsterism** [reply]
thanks guys. really excited for this !!

**Reply by mingsterism** [reply]
heyy guys. saw that 0.15.0 pull request was merged !!
would like to know is there a link we can go to for documentation to test it out? 
also shall we look at the docs under kitql or under houdini?

**Reply by AlecAivazis** [reply]
Wow you're quick! Yep! Everything is getting merged as I type this 🎉 we really appreciate everyone's patience while we figured out all the small details. 

We already have an update for the KitQL docs but @jycouet is speaking at Svelte Austria tonight so he has to finish preparing his slides. We'll update this discussion once everything is ready so you can check it out 👍

**Comment by jycouet** [comment]
@AlecAivazis and I are happy to unveil a massive effort

✅ New version of Houdini
✅ New version of KitQL

We are joining forces to:
> Bring the best GraphQL experience to Svelte 🎉

Infos & guides in the link below 👇
https://www.the-guild.dev/blog/houdini-and-kitql

---

## Now, we need **_YOU_**. 
Please make some noise on
- [Twitter](https://twitter.com/jycouet/status/1543979659125313537?s=20&t=vH6k5_vg2C31rBFfqcv4hg)
- [Discord](https://discord.com/channels/457912077277855764/837012201444999248/993595306807656478)
- [Reddit](https://www.reddit.com/r/sveltejs/comments/vrk7hc/houdini_and_kitql_join_forces_selfpromotion/)

🎉🥳

**Comment by maku** [comment]
Just to be curious. When I have a existing graphql backend should I use only houdini or kitql?

**Reply by jycouet** [reply]
Houdini, as it's the client, will do the perfect job for you 👍
